### PR TITLE
Drop chart for provider-local `ControllerRegistration` and deploy it directly using skaffold

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:

This PR drops some hacks we needed to make skaffold deploy the provider-local `ControllerDeployment` with the correct image reference.

With the latest version, we can teach skaffold to inject image references into custom resources as well via `resourceSelector`.
However, this requires the `image` field to be a flat field (i.e., fully-qualified image reference) instead of the usual `image` stanza with values for `repository` and `tag`.
To make this work, this PR first changes `generate-controller-registration.sh` to allow using a fully-qualified `image` field.

**Which issue(s) this PR fixes**:

Part of #6016 

**Special notes for your reviewer**:

The change is backward-compatible: if the extension chart has the `image.repository` value, the script will include the same values as before.
If the extension chart uses a flat `image` value, the script now includes only the `image` field in the output values.

This is a good starting point for adding a skaffold-based developer experience to extensions as well (`make extension-up`).

**Release note**:

```feature developer
`generate-controller-registration.sh` now supports extension charts with fully-qualified `image` values instead of the usual `image` stanza with values for `repository` and `tag`. With this, skaffold can be configured (using `resourceSelector`) to inject a freshly-built image reference into the generated `ControllerDeployment`.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -233,7 +233,7 @@ check: $(GO_ADD_LICENSE) $(GOIMPORTS) $(GOLANGCI_LINT) $(HELM) $(IMPORT_BOSS) $(
 	@hack/check-skaffold-deps.sh
 
 .PHONY: generate
-generate: $(CONTROLLER_GEN) $(GEN_CRD_API_REFERENCE_DOCS) $(GOIMPORTS) $(GO_TO_PROTOBUF) $(HELM) $(MOCKGEN) $(OPENAPI_GEN) $(PROTOC_GEN_GOGO) $(YAML2JSON)
+generate: $(CONTROLLER_GEN) $(GEN_CRD_API_REFERENCE_DOCS) $(GOIMPORTS) $(GO_TO_PROTOBUF) $(HELM) $(MOCKGEN) $(OPENAPI_GEN) $(PROTOC_GEN_GOGO) $(YAML2JSON) $(YQ)
```
Comment (plkokanov) [inline Makefile]: `$(YQ)` should also be specified for the `generate-sequential` make target, right?

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR! I just one  minor question and suggestion.

File: Makefile (hunk 1 above)
Comment (timebertt) [reply Makefile]: Good catch, thanks! I unified the dependencies of both `generate*` targets in a new commit so that no one will forget this again.

**Comment by timebertt** [comment]
@plkokanov I rebased the PR and addressed your suggestion. PTAL :)

**Comment by rfranzke** [comment]
Thanks, this looks great! Approving since @plkokanov's comment was rather minor and is addressed now.

/lgtm
/approve

**Pull request merged**